### PR TITLE
Correct metadata supported platforms

### PR DESCRIPTION
**Pull request by bmhughes**
# Description

- Correct metadata supported platforms

## Issues Resolved

n/a

## Check List

- [x] All tests pass. See TESTING.md for details.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable.

### CODE REVIEW

File: metadata.rb [hunk 1]
```diff
@@ -8,8 +8,11 @@
 chef_version      '>= 15'
 version           '5.0.1'
 
-supports 'ubuntu'
-supports 'redhat'
+supports 'amazon'
+supports 'debian'
 supports 'centos'
+supports 'redhat'
+supports 'suse'
```
Comment (ramereth) [inline metadata.rb:15]: Should this also include `opensuseleap`?

**Comment by ramereth** [comment]
@bmhughes this PR might be a good place to also add the missing changelog entries for the last couple of releases that were missed. Would you mind doing that?

File: metadata.rb (hunk 1 above)
Comment (bmhughes) [reply metadata.rb:15]: Indeed it should

**Review by ramereth — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [5.0.2](https://github.com/sous-chefs/vault/releases/tag/5.0.2)

**Pull request merged**